elastic_service/elastic_handler.go

```diff
@@ -0,0 +1,152 @@
+package elastic_service
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/daniyakubov/book_service_n/config"
+	"github.com/daniyakubov/book_service_n/consts"
+	"github.com/daniyakubov/book_service_n/db_service"
+	"github.com/daniyakubov/book_service_n/models"
+	errors "github.com/fiverr/go_errors"
+	"github.com/olivere/elastic/v7"
+	"strconv"
+	"strings"
+)
+
+var _ db_service.DbHandler = &ElasticHandler{}
+
+type ElasticHandler struct {
+	Url          string
+	Client       *elastic.Client
+	maxSizeQuery int
+}
+
+func NewElasticHandler(url string, client *elastic.Client, maxSizeQuery int) ElasticHandler {
+	return ElasticHandler{
+		url,
+		client,
+		maxSizeQuery,
+	}
+}
+
+func (e *ElasticHandler) Update(ctx *context.Context, title string, id string) (err error) {
+	_, err = e.Client.Update().
+		Index(consts.Index).
+		Id(id).
+		Doc(map[string]interface{}{consts.Title: title}).
+		Do(*ctx)
+	if err != nil {
+		return errors.Wrap(err, err.Error())
+	}
+	return nil
+}
+
+func (e *ElasticHandler) Add(ctx *context.Context, body []byte) (string, error) {
+	res, err := e.Client.Index().
+		Index(consts.Index).
+		BodyString(string(body)).
+		Do(*ctx)
+	if err != nil {
+		return "", errors.Wrap(err, err.Error())
+	}
+	return res.Id, err
+}
+func (e *ElasticHandler) Get(ctx *context.Context, id string) (src *models.Book, err error) {
+	get, err := e.Client.Get().
+		Index(consts.Index).
+		Id(id).
+		Do(*ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, err.Error())
+	}
+
+	var book models.Book
+	if err = json.Unmarshal(get.Source, &book); err != nil {
+		return nil, err
+	}
+	return &book, err
+}
+
+func (e *ElasticHandler) Delete(ctx *context.Context, id string) error {
+	_, err := e.Client.Delete().
+		Index(consts.Index).
+		Id(id).
+		Do(*ctx)
+	if err != nil {
+		return errors.Wrap(err, err.Error())
+	}
+	return err
```

Comment:
```suggestion
	return err
```

Or you can customize the err with relevant info
```suggestion
        if err != nil {
                  return errors.Wrap(err, fmt.Stringf("deleted book id: %s failed\n", id))
         }
	return nil
```